### ISSUE TITLE
помогите разобрать

### ISSUE THREAD

**Issue by yo8z6gv** [bol-van/zapret-win-bundle#29]

**Comment by bol-van** [comment]
> --filter-udp=50000-65535 --dpi-desync=fake,tamper --dpi-desync-any-protocol --dpi-desync-fake-quic="%~dp0quic_initial_www_google_com.bin"

Это ТРЭШ ФЛУД вариант, распространенный кем-то в сети. Везде отмечаю как таковой. Его использовать нельзя, потому что нет cutoff ограничителя. tamper бесполезен. Диапазон портов слишком широкий.

Вам надо понять, что профили разделяются параметром --new. Внутри профиля порядок указания параметров значения не имеет. Есть параметры фильтра, есть параметры дурения. Их можно мешать как угодно внутри профиля. Суть не меняется. И второе - проверка профилей идет до первого совпадения с начала до конца. Именно по порядку, а не иначе

**Comment by yo8z6gv** [comment]
я разделяю на несколько служб чтобы удобнее было управлять через диспетчер задач. 

для голосового дискорда хватило и
```
set ARGS=--wf-udp=50000-65535 ^
--filter-udp=50000-65535 --dpi-desync=fake --dpi-desync-any-protocol
call :srvinst winws1discord

goto :eof

:srvinst
net stop %1
sc delete %1
sc create %1 binPath= "\"%~dp0winws.exe\" %ARGS%" DisplayName= "zapret DPI bypass : %1" start= auto
sc description %1 "zapret DPI bypass software"
sc start %1
```
так что согласен что tamper бесполезен. но как тогда дискорд настроить, голосовой RTC

**Comment by bol-van** [comment]
--filter-udp=50000-50099 --dpi-desync=fake --dpi-desync-any-protocol --dpi-desync-cutoff=n2

**Comment by Anclave** [comment]
простите за нубский вопрос ...
Вот я прогнал blockcheck.cmd всякие варианты (кстати он там тока запрет на рутрекер пробует обойти? для ютуба какой сервер тестить?)
В логе он мне написал:
* SUMMARY
ipv4 rutracker.org curl_test_http : winws --wf-l3=ipv4 --wf-tcp=80 --dpi-desync=split --dpi-desync-ttl=10
ipv4 rutracker.org curl_test_https_tls12 : winws --wf-l3=ipv4 --wf-tcp=443 --dpi-desync=disorder2
ipv4 rutracker.org curl_test_http3 : winws --wf-l3=ipv4 --wf-udp=443 --dpi-desync=fake

я так понял это строчки для создания службы, но я хочу интерактивный запуск через preset_russia.cmd т.к. иногда что-то может не дружить с дпи...
Но в итоге какой батник на выходе должен быть?
Ведь там внутри надо хостлисты указывать и т.п. а я с синтаксисом этих настроек не сильно дружу и вообще изначально там 7 строчек всяких параметров.

**Comment by alk2git** [comment]
эти строчки не для создания службы. Из них надо взять нужные строчки, чтобы добавить в ваш bat-файл.
В ту его часть, которая идет со строчками, начинающимися с --filter. То есть в Вашем случае строки будут:
start "zapret: http,https,quic" /min "%~dp0winws.exe" ^
--wf-tcp=80,443 --wf-udp=443,50000-50099 ^
--filter-tcp=80 --wf-l3=ipv4 --wf-tcp=80 --dpi-desync=split --dpi-desync-ttl=10 --new ^
--filter-tcp=443 --dpi-desync=disorder2 --new ^
--filter-udp=443 --dpi-desync=fake
....

**Comment by yo8z6gv** [comment]
@Anclave нет. это настройки фильтра для winws. а тип запуска winws уже определяется windows.
т.е для служб это будет 
```
call :srvinst
net stop
sc delete
sc create
sc description
sc start
```

для задачи (**task**), типо cmd, но как приложение или я хз как это работает. команды:
```
schtasks /Create /F /TN 
schtasks /Run /TN 
```

**WINWS1** же переменная, которая и устанавливается этим
`set WINWS1 `

собственно полные командные строки так и выглядят,
`schtasks /Create /F /TN winws1 /NP /RU "" /SC onstart /TR "\"%~dp0winws.exe\" %WINWS1%"`
```
schtasks — инструмент для управления расписанием задач в Windows.
/Create — создаёт новую задачу.
/F — принудительно перезаписывает задачу, если она уже существует.
/TN winws1 — указывает имя задачи (winws1).
/NP — создаёт задачу без сохранения пароля (не использует аутентификацию).
/RU "" — запускает задачу от имени локальной системной учетной записи.
/SC onstart — указывает, что задача должна выполняться при запуске системы.
/TR "\"%~dp0winws.exe\" %WINWS1%" — определяет действие задачи: запускает программу winws.exe, расположенную в той же папке, что и скрипт, с передачей аргумента %WINWS1%.
```

ну и как я понимаю это все. других вариантов нет¯\_(ツ)_/¯
чтобы запустить от командой строки, пока активна - будет работать, нужно просто все аргументы указывать сразу в cmd.
без этих промежуточных task(задача) или service(служба)
```
start "zapret: http,https,autohostlist" /min "%~dp0winws.exe" --wf-tcp=80,443 --dpi-desync=fake,disorder2 --dpi-desync-autottl=2 --dpi-desync-fooling=md5sig --hostlist-auto="%~dp0autohostlist.txt"
start "zapret: quic,autohostlist" /min "%~dp0winws.exe" /min --wf-udp=443 --dpi-desync=fake --dpi-desync-repeats=11 --hostlist-auto="%~dp0autohostlist.txt"
```
в твоем случае нужно этот preset_my-conf-autohostlist.cmd создать с следующим содержанием:
```
start "zapret: http,https,autohostlist" /min "%~dp0winws.exe" --wf-tcp=80 --dpi-desync=split --dpi-desync-ttl=10 --hostlist-auto="%~dp0autohostlist.txt"
start "zapret: http,https,autohostlist" /min "%~dp0winws.exe" --wf-tcp=443 --dpi-desync=disorder2 --hostlist-auto="%~dp0autohostlist.txt"
start "zapret: quic,autohostlist" /min "%~dp0winws.exe" /min --wf-udp=443 --dpi-desync=fake --hostlist-auto="%~dp0autohostlist.txt"
```
первая строка "`zapret: http,https,autohostlist`" это название батника, остальное, означает в трее запуск? /min, затем что запускается "`%~dp0winws.exe`" - из текущего каталога файл `winws.exe`, с переменным `--wf-tcp=80 --dpi-desync=split --dpi-desync-ttl=10 --hostlist-auto="%~dp0autohostlist.txt"`
на всякий я разделил несколькими строками `dpi-desync=split` и `dpi-desync=disorder2` как у тебя в примере, тк не знаю, совместимо ли все вместе в одной строке.

тк допустим на win11 такой вариант батника с переменной не запустится,
`--filter-udp=443 --dpi-desync=fake --hostlist="%~dp0autohostlist.txt"`
нельзя использовать `--hostlist="%~dp0autohostlist.txt"` ¯\_(ツ)_/¯
